providers/catalog.go

```diff
@@ -79,6 +80,36 @@ const (
 // ================================================================================
 
 var catalog = CatalogType{ // nolint:gochecknoglobals
+
+	// Adobe Sign configuration with substitution
+	AdobeSign: {
+		AuthType: Oauth2,
+		BaseURL:  "https://api.{{.shard}}.adobesign.com",
```

Comment:
Can you also add the endpoint to call for fetching the shard, and a sample response?

Reply:
@RajatPawar, I guess the flow has the two branches for customers and partners respectively (the branch/app type is set upon the app is created). 

- If you use the customer app you would need to specify shard in your Auth URL: https://secure.{{shard}}.echosign.com/public/oauth/v2 
- If you use the partner app you don't need to write a shard in Auth URL: https://secure.echosign.com/public/oauth/v2

Customer and partner apps both share the same  access token URL: https://secure.adobesign.com/oauth/v2/token 

Attaching a sceenshot of a token for customer as the code we have is for customer. Please let me know if you want additional code/connector to handle the partner use case. 
![shard](https://github.com/amp-labs/connectors/assets/95291462/a751df81-af67-4265-9225-b56d2d017708)